workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/UnityComponentDetails.cs

```diff
@@ -51,5 +52,33 @@ public void PopulateFields(DetailsStore store)
                     .AsReadOnly();
             }
         }
+
+        public bool IsValid()
+        {
+            var isValid = true;
+            var componentName = ComponentName;
+
+            var clashingCommands = CommandDetails
+                .Where(commandDetail => commandDetail.CommandName.Equals(componentName));
+            foreach (var clashingCommand in clashingCommands)
+            {
+                isValid = false;
+                Console.Error.WriteLine(
+                    $"Error in component \"{componentName}\". " +
+                    $"Command \"{clashingCommand.RawCommandName}\" clashes with component name.");
```

Comment:
Nit: Prefer having a long line over concatentating strings. Also you can use single quotes and get rid of the escaping which aids readability

Reply:
I was using single quotes before and I think it's not really that readable in the Editor